src/hb-subset-plan.cc

```diff
@@ -209,15 +211,46 @@ _fill_unicode_and_glyph_map(hb_subset_plan_t *plan,
   _fill_unicode_and_glyph_map(plan, unicode_iterator, unicode_to_gid_for_iterator, unicode_to_gid_for_iterator);
 }
 
+/*
+ * Finds additional unicode codepoints which are reachable from the input unicode set.
+ * Currently this adds in mirrored variants (needed for bidi) of any input unicodes.
+ */
+static hb_set_t
+_unicode_closure (const hb_set_t* unicodes, bool bidi_closure) {
+  // TODO: we may want to also consider pulling in reachable unicode composition and decompositions.
+  //       see: #2283
+  hb_set_t out = *unicodes;
+  if (!bidi_closure) return out;
+
+  if (out.is_inverted()) {
+    // don't closure inverted sets, they are asking to specifically exclude certain codepoints.
+    // otherwise everything is already included.
+    return out;
+  }
+
+  for (hb_codepoint_t cp : *unicodes) {
+   hb_codepoint_t mirror = hb_unicode_mirroring(hb_unicode_funcs_get_default (), cp);
+   if (mirror != cp) {
+     out.add(mirror);
+   }
+   out.add(cp);
```

Comment:
Why add `cp`? Isn't `out` already a copy of `unicodes`?

Reply:
Oops, reworked that while writing the code and forgot to remove the add. Fixed.